### DISCUSSION TITLE
Software Keyboard

### DISCUSSION THREAD

**Discussion by ObscenityIB** [uyjulian/renpy-switch#20] [Q&A]
How can I go about using the libnx keyboard applet?

I think that will fix a bug where the following code causes a keyboard loop, it keeps opening after being closed.

```
unk "\"Whats your name?\""
$ pname = renpy.input("Whats your name?")
$ pname = pname.strip()
if pname == "":
        $ pname="Unset"
```

**Comment by DebugDax** [comment]
An example provided by someone who knows more about how to use the libnx bindings would be great.
I found the Keyboard function list here: https://github.com/switchbrew/libnx/blob/e2c3ae8e4461cbdfd3757304c75b41b6c551dc2e/nx/source/applets/swkbd.c#L269

I think swkbdShow is what's needed but to use it you would have to build a swkbdConfig (https://github.com/switchbrew/libnx/blob/e2c3ae8e4461cbdfd3757304c75b41b6c551dc2e/nx/include/switch/applets/swkbd.h#L174) then handle the return value, that's where i'm lost at.

_Maybe_ something like this? But then another issue is being how to pass size_t of out_string_size?
```
import libnx
define inputVariable

swKeyboardConfig = libnx.SwkbdConfig()
libnx.swkbdShow(swKeyboardConfig, inputVariable, out_string_size)
```

To summize, I have no clue and would also like to figure this out. Maybe my guesswork can lead someone else in the proper direction to guide us.

**Comment by ObscenityIB** [comment]
hey @HotdogCombo anything come up?
